### PR TITLE
All public members have detailed documentation on what they do/are

### LINKED ISSUE #55: Minor: Improve Method Documentation Throughout
A lot of methods need documentation for what they do, how, and why. Examples:
* Missing XML summaries on methods in WebApi.NotificationHub
* DataAccess.NotificationContext only has basic comments

### PR DESCRIPTION
**Pull request by ibnetahir1**
The DataAccess, Domain, and WebApi projects now have more detailed documentations for all members of each class. Solves issue #55.

### CODE REVIEW

**Review by jpw547-zz — COMMENTED** [review]
The documentation looks good to me!

File: FakebookNotifications/FakebookNotifications.DataAccess/Models/Notification.cs [hunk 1]
```diff
@@ -4,30 +4,47 @@
...
+        /// Notification Id in the database
+        /// </summary>
         [BsonId]
         [BsonRepresentation(BsonType.ObjectId)]
         public string Id { get; set; }
 
-        //notification type: post/comment/tag  
+        /// <summary>
+        /// The notification type (post, comment, like, follow)
+        /// </summary>
```
Comment (escalonn) [inline FakebookNotifications/FakebookNotifications.DataAccess/Models/Notification.cs:24]: clarify the meaning of the `int` here.

File: FakebookNotifications/FakebookNotifications.DataAccess/Models/NotificationsContext.cs [hunk 2]
```diff
@@ -43,11 +48,21 @@ public NotificationsContext(IOptions<NotificationsDatabaseSettings> settings, IL
...
+        /// <summary>
+        /// Gets the notification collection from the database
+        /// </summary>
         public IMongoCollection<Notification> Notifications => _database.GetCollection<Notification>(_settings.NotificationsCollection);
+
+        /// <summary>
+        /// Deletes all notifications in the notification collection
+        /// </summary>
+        /// <param name="noteCol">The notification collection</param>
+        /// <returns></returns>
```
Comment (escalonn) [inline FakebookNotifications/FakebookNotifications.DataAccess/Models/NotificationsContext.cs]: document return value

File: FakebookNotifications/FakebookNotifications.Domain/Models/Notification.cs [hunk 3]
```diff
@@ -3,13 +3,41 @@
...
     {
+
+        /// <summary>
+        /// Id of the notification
+        /// </summary>
         public string Id { get; set; }
+
+        /// <summary>
+        /// The type of notification it is. The key is a string that states the type (comment, follow, like).
+        /// </summary>
```
Comment (escalonn) [inline FakebookNotifications/FakebookNotifications.Domain/Models/Notification.cs:21]: again, document what the int means

File: FakebookNotifications/FakebookNotifications.WebApi/FakebookNotifications.WebApi.csproj [hunk 4]
```diff
@@ -2,6 +2,7 @@
 
   <PropertyGroup>
     <TargetFramework>net5.0</TargetFramework>
+    <UserSecretsId>c5cf906f-d5db-4953-9447-58d487547d5b</UserSecretsId>
```
Comment (escalonn) [inline FakebookNotifications/FakebookNotifications.WebApi/FakebookNotifications.WebApi.csproj:5]: this is fine, but i'm curious what you're using user secrets for here

File: FakebookNotifications/FakebookNotifications.WebApi/Hubs/NotificationHub.cs [hunk 5]
```diff
@@ -179,19 +197,34 @@ public async Task SendAll(string user, string notification)
...
         }
 
-        //Send a notification to all clients within a specific group
+        /// <summary>
+        /// Send a notification to all clients within a specific group. Can be used to send a notification to
+        /// all connected devices of the same user.
+        /// </summary>
+        /// <param name="group">group name to send notifications to</param>
+        /// <param name="notification">notification to be sent as string</param>
+        /// <returns></returns>
```
Comment (escalonn) [inline FakebookNotifications/FakebookNotifications.WebApi/Hubs/NotificationHub.cs]: there's boilerplate we can put here for any Task returns, run a search for "c# xml documentation Task async returns" or something like that

File: FakebookNotifications/FakebookNotifications.WebApi/Hubs/NotificationHub.cs [hunk 6]
```diff
@@ -214,8 +247,9 @@ public async Task SendUserGroupAsync(Domain.Models.User user, Domain.Models.Noti
                 throw new NullReferenceException();
             }
         }
+
         /// <summary>
-        /// Method to send notifcications to every active connection of a given user
+        /// Sends notifcications to every active connection of a given user
```
Comment (escalonn) [inline FakebookNotifications/FakebookNotifications.WebApi/Hubs/NotificationHub.cs]: fix the typo

**Review by escalonn — CHANGES_REQUESTED** [review]

**Comment by escalonn** [comment]
if we know all the notification types we expect, we should document that somewhere (not to mention validate it somewhere, but in a different PR)

File: FakebookNotifications/FakebookNotifications.WebApi/FakebookNotifications.WebApi.csproj (hunk 4 above)
Comment (ibnetahir1) [reply FakebookNotifications/FakebookNotifications.WebApi/FakebookNotifications.WebApi.csproj:5]: nothing was added, I clicked on "Manage user secrets" to see what was there and it just created that.

**Review by jpw547-zz — APPROVED** [review]
Things should be good now

**Review by escalonn — APPROVED** [review]
technically the user secrets change doesn't belong here, but i'll let it through

**Pull request merged**